miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -243,87 +243,72 @@ end
 
 #! Creates a new note and returns a pointer to the memory address at which the note is stored.
 #!
-#! Inputs: [ASSET, tag, note_type, RECIPIENT]
-#! Outputs: [note_ptr, 0, 0, 0, 0, 0, 0, 0, 0]
+#! Inputs: [tag, note_type, RECIPIENT]
+#! Outputs: [note_ptr]
 #!
-#! ASSET is the asset to be included in the note.
 #! tag is the tag to be included in the note.
 #! RECIPIENT is the recipient of the note.
-#! ptr is the pointer to the memory address at which the note is stored.
+#! note_ptr is the pointer to the memory address at which the note is stored.
+#!
+#! Panics if
+#! - the note_type is not valid
+#! - the note_tag is not an u32
```

Comment:
There are a couple more failure conditions:

- `note_type` and `note_tag` are inconsistent (if `note_tag` starts with anything but `0b11`, note must be public).
- We are trying to create too many notes (I believe the current limit is 1024 per transaction?). Though, we should check if this is actually being enforced in `increment_num_created_notes`.